go/cmd/dolt/cli/stdio.go

```diff
@@ -143,6 +140,37 @@ func PrintErrf(format string, a ...interface{}) {
 	fmt.Fprintf(CliErr, format, a...)
 }
 
+func DeleteAndPrint(prevMsgLen int, msg string) int {
+	if outputIsClosed() {
+		return 0
+	}
+
+	msgLen := len(msg)
+	backspacesAndMsg := make([]byte, prevMsgLen+msgLen, 2*prevMsgLen+msgLen)
```

Comment:
Is `2*prevMsgLen+msgLen` the exact length or are just approximating to try and get close? It seems like it would be something more like prevMsgLen+msgLen+2*diff ?

Reply:
It's an estimate! That value is for the capacity of the buffer to prevent the `append` calls from needing to grow the capacity too often.

From go documentation:
> The append function appends the elements x to the end of the slice s, and grows the slice if a greater capacity is needed.